app/screens/integration_selector/integration_selector.tsx

```diff
@@ -190,56 +202,23 @@ function IntegrationSelector(
 
         switch (dataSource) {
             case ViewConstants.DATA_SOURCE_USERS: {
-                setMultiselectSelected((current) => {
-                    const multiselectSelectedItems = {...current};
-
-                    if (current[itemKey]) {
-                        delete multiselectSelectedItems[itemKey];
-                    } else {
-                        multiselectSelectedItems[itemKey] = item as UserProfile;
-                    }
-
-                    return multiselectSelectedItems;
-                });
+                setMultiselectSelected((current) => toggleFromMap(current, itemKey, item as UserProfile));
                 return;
             }
             case ViewConstants.DATA_SOURCE_CHANNELS: {
-                setMultiselectSelected((current) => {
-                    const multiselectSelectedItems = {...current};
-
-                    if (current[itemKey]) {
-                        delete multiselectSelectedItems[itemKey];
-                    } else {
-                        multiselectSelectedItems[itemKey] = item as Channel;
-                    }
-
-                    return multiselectSelectedItems;
-                });
+                setMultiselectSelected((current) => toggleFromMap(current, itemKey, item as Channel));
                 return;
             }
             default: {
-                setMultiselectSelected((current) => {
-                    const multiselectSelectedItems = {...current};
-
-                    if (current[itemKey]) {
-                        delete multiselectSelectedItems[itemKey];
-                    } else {
-                        multiselectSelectedItems[itemKey] = item as DialogOption;
-                    }
-
-                    return multiselectSelectedItems;
-                });
+                setMultiselectSelected((current) => toggleFromMap(current, itemKey, item as DialogOption));
             }
         }
     }, [isMultiselect, dataSource, handleSelect]);
 
     const handleRemoveOption = useCallback((item: UserProfile | Channel | DialogOption) => {
-        const currentSelected: Dictionary<UserProfile> | Dictionary<DialogOption> | Dictionary<Channel> = multiselectSelected;
         const itemKey = extractItemKey(dataSource, item);
-        const multiselectSelectedItems = {...currentSelected};
-        delete multiselectSelectedItems[itemKey];
-        setMultiselectSelected(multiselectSelectedItems);
-    }, [dataSource, multiselectSelected]);
+        setMultiselectSelected((current) => toggleFromMap(current, itemKey, item));
```

Comment:
My idea was not exactly to use `toggleFromMap`, just the `setX((cur) => {...})` shape of the setState. This kind of works, but it would be safer to keep more or less the logic we used to have, but inside the `setX((cur) => {...})` function.

Reply:
ok! I thought it was nicer to have the same, since in both we are deleting an item by key from the map.

Reply:
You are right, and in general it should be safe, but I still prefer to lean on the "even safer" side.

Imagine that for whatever reason, this is called twice, or it is called when the item is not selected. That way we make sure this does in any case the behaviour it is expected to do.